### PR TITLE
Add Authorization enum to decide between api-key header or authorization header

### PR DESCRIPTION
**Pull request by mfreiwald**
As mentioned in #12 Azure OpenAI has `api-key` as header field for the raw api-key access.\
`Authorization` header is used for access tokens received from Microsoft Entra ID logins.

This PR adds an extra enum `Authorization` to decide which header field should be used when connecting to the OpenAI services.

### CODE REVIEW

File: Sources/OpenAI/Public/Service/DefaultOpeanAIService.swift [hunk 1]
```diff
@@ -14,14 +14,14 @@ struct DefaultOpenAIService: OpenAIService {
...
-   private let apiKey: String
+   private let apiKey: Authorization
    /// [organization](https://platform.openai.com/docs/api-reference/organization-optional)
    private let organizationID: String?
    
    private static let assistantsBeta = "assistants=v1"
       
    init(
-      apiKey: String,
+      apiKey: Authorization,
```
Comment (mfreiwald) [inline Sources/OpenAI/Public/Service/DefaultOpeanAIService.swift:24]: We could add an `assert`, so `DefaultOpenAIService` can only instantiated with a `.bearer` Authorization

Comment (jamesrochabrun) [reply Sources/OpenAI/Public/Service/DefaultOpeanAIService.swift:24]: Thanks! wondering if you could use this instead https://github.com/jamesrochabrun/SwiftOpenAI?tab=readme-ov-file#azure-openai-1
`DefaultOpenAIAzureService` should work with Azure although now just supports Completions API. 
Let me know if that is what you need!

Comment (mfreiwald) [reply Sources/OpenAI/Public/Service/DefaultOpeanAIService.swift]: I changed it back so the default OpenAI API will always use bearer authorization.

**Comment by mfreiwald** [comment]
This is how the SDK can be used after this PR

```swift
// Use OpenAI API with API-Key string
let service = OpenAIServiceFactory.service(apiKey: apiKey)

// API-Key from the Azure resource group
let azureAPIKey = AzureOpenAIConfiguration(
                           resourceName: "YOUR_RESOURCE_NAME", 
                           openAIAPIKey: .apiKey("YOUR_OPENAI_APIKEY), 
                           apiVersion: "THE_API_VERSION")

// access token from a Microsoft Entra ID Login
let azureAccessToken = AzureOpenAIConfiguration(
                           resourceName: "YOUR_RESOURCE_NAME", 
                           openAIAPIKey: .bearer("YOUR_ACCESS_TOKEN), 
                           apiVersion: "THE_API_VERSION")

let azureService = OpenAIServiceFactory.service(azureConfiguration: AzureOpenAIConfiguration)

```

**Review by jamesrochabrun — APPROVED** [review]
Thanks!

**Pull request merged**